client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -1337,18 +1337,30 @@ Params withRealtime(boolean realtime) {
 
         Params withRefresh(boolean refresh) {
             if (refresh) {
-                return withRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+                return withRefreshPolicy(RefreshPolicy.IMMEDIATE);
             }
             return this;
         }
 
+        /**
+         *  @deprecated If creating a new HLRC ReST API call, use {@link RefreshPolicy}
+         *  instead of {@link WriteRequest.RefreshPolicy} from the server project
+         */
+        @Deprecated
```

Comment:
isn't it odd that the existing oss methods use and still need the deprecated variant, and there is no alternative to it?

Reply:
Isn't that the standard we've adopted by making the HLRC performRequest methods deprecated that take the existing request classes in #32912? The alternative is available by swapping to the POJO request/response classes that we want the HLRC to move towards?

Reply:
Yea but I think there's a slight difference, but I may be missing something. The deprecations we added until now are not directly visible to users while this one is, because users are specifying the policy and they at the moment can only specify the deprecated one.

Reply:
This isn't user facing code. It is only used in the `RequestConverters` class to convert a request into XContent with parameters. Users should really only be interacting with the client, request, and response objects. I could also be missing something